applications/OptimizationApplication/custom_utilities/filtering/explicit_filter_utils.cpp

```diff
@@ -101,82 +80,75 @@ double GetDomainSize(
     return pExpression->Evaluate(rPoint.Id(), rPoint.Id(), 0);
 }
 
-struct NonIntegratedWeight
+struct MeshIndependentType
 {
     static double Compute(
-        const double FilterFunctionWeight,
+        const double Value,
         const double DomainSize)
     {
-        return FilterFunctionWeight;
+        return Value;
     }
 };
 
-struct IntegratedWeight
+struct MeshDependentType
 {
     static double Compute(
-        const double FilterFunctionWeight,
+        const double Value,
         const double DomainSize)
     {
-        return FilterFunctionWeight * DomainSize;
+        return Value / DomainSize;
     }
 };
 
-template <class TEntityType, class TWeightComputationType>
+template <class TEntityType>
 void ComputeWeightForAllNeighbors(
```

Comment:
Hi Reza, Now, the damping coefficients (D matrix) is computed somewhere else, and provided to the explicit filter. This is changed in this manner to allow different dampinh matrix computations to use the same filtering mechanicsm (at the moment, damping finds the neareset dampes node, and calculate the weights, this gives checkerboard pattern at the corners. So we have plans to implement new damping matrix computaion methods which does not give chekerboard patterns. For that this has to be changed).

I completely disagree with the performance issue. Because, in current and master implementations, both has search tree calls for filter neighbours and damping neareset neighbour search. Therefore, they are in par with the performance. Because, now D matrix is passed as an expression.

As I see, this is more generic, than being specific. In the case of filterig, explicit filters can have different damping matrices which is a nice feature I want to allow.